base/src/main/kotlin/net/corda/v5/base/types/MemberX500Name.kt

```diff
@@ -16,32 +16,120 @@ import javax.security.auth.x500.X500Principal
+ * Example usages:
+ *
+ * ```java
+ * String commonName = "commonName";
+ * String organisationUnit = "organisationUnit";
+ * String organization = "organization";
+ * String locality = "London";
+ * String state = "state";
+ * String country = "GB";
+ *
+ *
+ * MemberX500Name exampleNameFirst = new MemberX500Name(organization, locality, country);
+ * MemberX500Name exampleNameSecond = new MemberX500Name(commonName, organisationUnit, organization, locality, state, country);
+ * MemberX500Name exampleNameThird = new MemberX500Name(commonName, organization, locality, country);
+ *
+ * String commonNameForExampleNameThird = exampleNameThird.getCommonName();
+ * String organisationUnitForExampleNameThird = exampleNameThird.getOrganisationUnit();
+ * String organisationForExampleNameThird = exampleNameThird.getOrganisation();
+ * String localityForExampleNameThird = exampleNameThird.getLocality();
+ * String stateForExampleNameThird = exampleNameThird.getState();
+ * String countryForExampleNameThird = exampleNameThird.getCountry();
+ * X500Principal principalForExampleNameThird = exampleNameThird.getX500Principal();
+ *
+ * String name = "O=organization,L=London,C=GB";
+ * X500Principal principalForNewName = new X500Principal(name);
+ * MemberX500Name nameByPrincipal = MemberX500Name.build(principalForNewName);
+ * MemberX500Name nameByParse = MemberX500Name.parse(name);
+ *
+ * Map<String, String > map = MemberX500Name.toAttributesMap("CN=alice, OU=Accounting, O=R3, L=Seattle, ST=Washington, C=US");
+ * ```
+ *
+ * ```kotlin
+ * val commonName = "commonName"
+ * val organisationUnit = "organisationUnit"
+ * val organization = "organization"
+ * val locality = "London"
+ * val state = "state"
+ * val country = "GB"
+ *
+ * val exampleNameFirst = MemberX500Name(organization, locality, country)
+ * val exampleNameSecond = MemberX500Name(commonName, organisationUnit, organization, locality, state, country)
+ * val exampleNameThird = MemberX500Name(commonName, organization, locality, country)
+ *
+ * val commonNameForExampleNameThird = exampleNameThird.commonName
+ * val organisationUnitForExampleNameThird = exampleNameThird.organisationUnit
+ * val organisationForExampleNameThird = exampleNameThird.organisation
+ * val localityForExampleNameThird = exampleNameThird.locality
+ * val stateForExampleNameThird = exampleNameThird.state
+ * val countryForExampleNameThird = exampleNameThird.country
+ * val principalForExampleNameThird = exampleNameThird.x500Principal
+ * val name = "O=organization,L=London,C=GB"
+ * val principalForNewName = X500Principal(name)
+ * val nameByPrincipal = MemberX500Name.build(principalForNewName)
+ * val nameByParse = MemberX500Name.parse(name)
+ *
+ * val map = MemberX500Name.toAttributesMap("CN=alice, OU=Accounting, O=R3, L=Seattle, ST=Washington, C=US")
+ * ```
+ *
+ * @author Alexey Kadyrov
+ * @since DP2
+ *
+ * @param commonName Optional name by which the entity is usually known. Used only for services (for
```

Comment:
Not clear what is `service` and what is `organisation` and how they fit the member name. There should be some explanation and reference how to find that information.